src/librustc_mir/borrow_check/nll/region_infer/mod.rs

```diff
@@ -444,21 +439,30 @@ impl<'tcx> RegionInferenceContext<'tcx> {
         }
     }
 
+    pub(crate) fn compute_causal_info(&self, mir: &Mir<'tcx>) -> RegionCausalInfo {
```

Comment:
I think a comment would be nice:

---

Re-execute the region inference, this time tracking causal information. This is significantly slower, so it is done only when an error is being reported.